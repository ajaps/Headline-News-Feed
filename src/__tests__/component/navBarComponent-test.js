import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import NavBar from '../../components/HeadlineNavbar.jsx';

const category = [{ id: 'all', name: 'All' }];

describe('NavBar', () => {
let app;
  it('should render as expected', () => {
    app = shallow(<NavBar category={category}/>);
    const tree = toJson(app);
    expect(tree).toMatchSnapshot();
  });

  it('contains English language clickable element', () => {
    let app = shallow(<NavBar />);
    expect(app.exists('<a>Top</a>')).toBe(true);
  });
});