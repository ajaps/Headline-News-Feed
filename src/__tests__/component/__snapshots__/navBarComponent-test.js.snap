// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render as expected 1`] = `
<div
  className=" container navbar2"
>
  <div
    className="masthead"
  >
    <h3
      className="text-muted"
    >
      Category
    </h3>
    <nav>
      <ul
        className="nav nav-justified"
      >
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="all"
          name="All"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="business"
          name="Business"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="entertainment"
          name="Entertainment"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="gaming"
          name="Gaming"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="general"
          name="General"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="music"
          name="Music"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="politics"
          name="Politics"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="science-and-nature"
          name="Science"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="sport"
          name="Sport"
        />
        <CategoryComponent
          category={
            Array [
              Object {
                "id": "all",
                "name": "All",
              },
            ]
          }
          id="technology"
          name="Technology"
        />
      </ul>
    </nav>
  </div>
</div>
`;
